### ISSUE TITLE
💅 lint/style/useSelfClosingElements false positive error

### ISSUE THREAD

**Issue by 7flash** [biomejs/biome#3364]
### Environment information

```bash
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.1.0"
```

### Rule name

lint/style/useSelfClosingElements

### Playground link

N/A

### Expected result

When I run

npx @biomejs/biome lint ./src/index.tsx

its offering this fix which is wrong because script tag cannot be self-enclosing

<img width="1070" alt="Screenshot 2024-07-07 at 16 06 37" src="https://github.com/biomejs/biome/assets/4569866/2fcfb730-c236-4aee-864e-8ef2c8f69e8b">

318     │ - ········<script·src="/filepond/filepond.js"></script>
        318 │ + ········<script·src="/filepond/filepond.js"·/>

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
> script tag cannot be self-enclosing

I think this is only true in HTML?

But this rule is only enabled in JSX, and JSX transpiler should transpile them into the same code, whether you use self-closing elements or not. You can check it in the [babel playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=false&spec=false&loose=false&code_lz=MYewdgzgLgBAwiAtgB3AUzFAgjAvDACgEo8A-QgKAEgAeCYAJwEtlYIHhcAiAegDMmAGzSowAE35CR4MQDoAVhC6kaPes1akKRANwUKoSLAQp0mAEJ5CJXOQIUYjmHUYs2HbpOGiJA7zIUlGB4tXQogA&debug=false&forceAllTransforms=true&modules=umd&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=env%2Creact&prettier=false&targets=&version=7.24.7&externalPlugins=&assumptions=%7B%7D).

Did I miss something? Or what kind of issues did you actually run into after you applied the fixes?

**Comment by ematipico** [comment]
Good call, @Sec-ant; you're right.
